mplaltair/_convert.py

```diff
@@ -1 +1,167 @@
+import matplotlib
+import altair
 
+from ._data import _normalize_data
+
+def _process_data_mappings(f):
+    """Decorator for processing the data inside the encoding spec
+
+    This is meant to resolve aggregates and other artefacts
+    """
+    def g(enc_spec, data):
+        # FIXME: This should be replaced by a proper aggregation handling
+        if enc_spec.get("aggregate"):
+            raise NotImplementedError
+        
+        return f(enc_spec, data)
+    return g
+
+def _allowed_ranged_marks(enc_channel, mark):
+    """TODO: DOCS
+    """
+    return mark in ['area', 'bar', 'rect', 'rule'] if enc_channel in ['x2', 'y2'] else True
+
+def _get_column(enc_spec, data):
+    try:
+        field = enc_spec.get('field')
+        if not field:
+            raise ValueError("Field not specified in the encoding x")
+        return data[enc_spec['field']]
+    except KeyError:
+        raise KeyError("{} does not match any column in the data".format(enc_spec))
+
+@_process_data_mappings
+def _process_x(enc_spec, data):
+    """Returns the MPL encoding equivalent for Altair x channel
+    """
+    dx = _get_column(enc_spec, data)
+    return ('x', dx)
+
+@_process_data_mappings
+def _process_y(enc_spec, data):
+    """Returns the MPL encoding equivalent for Altair y channel
+    """
+    dy = _get_column(enc_spec, data)
+    return ('y', dy)
+
+@_process_data_mappings
+def _process_x2(enc_spec, data):
+    """Returns the MPL encoding equivalent for Altair x2 channel
+    """
+    raise NotImplementedError
+
+@_process_data_mappings
+def _process_y2(enc_spec, data):
+    """Returns the MPL encoding equivalent for Altair y2 channel
+    """
+    raise NotImplementedError
+
+@_process_data_mappings
+def _process_color(enc_spec, data):
+    """Returns the MPL encoding equivalent for Altair color channel
+    """
+    dc = _get_column(enc_spec, data)
+
+    dtype = enc_spec['type']
+    if dtype == 'quantitative':
+        return ('c', dc)
+    elif dtype == 'nominal':
+        raise NotImplementedError
+    elif dtype == 'ordinal':
+        return ('c', dc)
+    elif dtype == 'temporal':
+        raise NotImplementedError
+
+@_process_data_mappings
+def _process_fill(enc_spec, data):
+    """Returns the MPL encoding equivalent for Altair fill channel
+    """
+    return _process_color(enc_spec, data)
+
+@_process_data_mappings
+def _process_shape(enc_spec, data):
+    """Returns the MPL encoding equivalent for Altair shape channel
+    """
+    raise NotImplementedError
+
+@_process_data_mappings
+def _process_opacity(enc_spec, data):
+    """Returns the MPL encoding equivalent for Altair opacity channel
+    """
+    raise NotImplementedError
+
+@_process_data_mappings
+def _process_size(enc_spec, data):
+    """Returns the MPL encoding equivalent for Altair size channel
+    """
+    ds = _get_column(enc_spec, data)
+
+    dtype = enc_spec['type']
+    if dtype == 'quantitative':
```

Comment:
It may be better to do this as a `return some_mapping[dtype]` following the pattern used in `convert`.